src/main/java/com/o19s/es/ltr/ranker/parser/LightGBMJsonParser.java

```diff
@@ -0,0 +1,275 @@
+        static private final Split.Comparer COMPARER_SIMPLE = (threshold, fval) -> fval <= threshold;
+
+        public Split.Comparer numericalDecision() {
+            if (MissingType.None.equals(missingType) || (MissingType.Zero.equals(missingType) && defaultLeft && ZERO_THRESHOLD < threshold)) {
+                return COMPARER_SIMPLE;
+            }
+            if (MissingType.Zero.equals(missingType)) {
+                if (defaultLeft) {
+                    return COMPARER_ZERO_DEFAULT_LEFT;
+                } else {
+                    return COMPARER_ZERO_DEFAULT_RIGHT;
+                }
+            }
+            if (defaultLeft) {
+                return COMPARER_DEFAULT_LEFT;
+            } else {
+                return COMPARER_DEFAULT_RIGHT;
+            }
+        }
+
+        // Setters for all parsed fields
+
+        void setDefaultLeft(boolean defaultLeft) {
+            this.defaultLeft = defaultLeft;
+        }
+
+        void setMissingType(String missingType) {
+            setMissingType(MissingType.valueOf(missingType));
+        }
+
+        void setMissingType(MissingType missingType) {
+            this.missingType = missingType;
+        }
+
+        void setDecisionType(String decisionType) {
+            this.decisionType = decisionType;
+        }
+
+        void setThreshold(float threshold) {
+            this.threshold = threshold;
+        }
+
+        void setSplitFeature(int splitFeature) {
+            this.splitFeature = splitFeature;
+        }
+
+        void setRightChild(SplitParserState child) {
+            rightChild = child;
+        }
+
+        void setLeftChild(SplitParserState child) {
+            leftChild = child;
+        }
+
+        void setLeafValue(float leafValue) {
+            this.leafValue = leafValue;
+        }
+
+        // post-parse validation
+
+        boolean isSplit() {
+            return leafValue == null;
+        }
+
+        boolean hasAllSplitFields() {
+            return defaultLeft != null && missingType != null && decisionType != null && !Float.isNaN(threshold)
+                    && splitFeature >= 0 && rightChild != null && leftChild != null;
+        }
+
+        NaiveAdditiveDecisionTree.Node toNode(List<String> featureNames, FeatureSet set) {
+            if (isSplit()) {
+                String featureName = featureNames.get(splitFeature);
+                if (!set.hasFeature(featureName)) {
+                    // Can't throw parsing exceptions anymore...
+                    throw new RuntimeException("TODO");
+                }
+                int ordinal = set.featureOrdinal(featureName);
+                NaiveAdditiveDecisionTree.Node left = leftChild.toNode(featureNames, set);
+                NaiveAdditiveDecisionTree.Node right = rightChild.toNode(featureNames, set);
+                return new Split(left, right, set.featureOrdinal(featureName), threshold, numericalDecision());
```

Comment:
already have the ordinal in a variable, no need to ask again (or remove the variable declaration)